### PR TITLE
Implemented dpnp.hstack() and dpnp.atleast_1d() functions.

### PR DESCRIPTION
**Pull request by npolina4**
Implemented dpnp.hstack() and dpnp.atleast_1d() functions.
Added tests.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_manipulation.py [hunk 1]
```diff
@@ -476,14 +514,62 @@ def expand_dims(a, axis):
...
+    array([[1, 4],
+           [2, 5],
+           [3, 6]])
+
     """
 
+    if casting != "same_kind":
+        pass
+    elif dtype is not None:
+        pass
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_manipulation.py:562]: can we implement support of all parameters through existing calls, like for `dpnp.copyto` is done and so to get rid of `call_origin` completely or would be easier to left this when full support of parameters done in `dpnp.concatenate`?

File: tests/test_arraymanipulation.py [hunk 2]
```diff
@@ -605,3 +601,10 @@ def test_2D_array2(self):
     def test_generator(self):
         with assert_warns(FutureWarning):
             dpnp.vstack((numpy.arange(3) for _ in range(2)))
+
+
+@pytest.mark.parametrize("arys", [1.0, dpnp.arange(9.0).reshape(3, 3), [3, 4]])
```
Comment (antonwolfy) [inline tests/test_arraymanipulation.py:606]: please remove `allow_fall_back_on_numpy` fixture from `tests/third_party/cupy/manipulation_tests/test_dims.py::TestDims::test_atleast_1d1` and `tests/third_party/cupy/manipulation_tests/test_dims.py::TestDims::test_atleast_1d2`

File: tests/test_arraymanipulation.py [hunk 3]
```diff
@@ -605,3 +601,10 @@ def test_2D_array2(self):
     def test_generator(self):
         with assert_warns(FutureWarning):
             dpnp.vstack((numpy.arange(3) for _ in range(2)))
+
+
```
Comment (antonwolfy) [inline tests/test_arraymanipulation.py:605]: please update tests in `tests/third_party/cupy/manipulation_tests/test_join.py`

File: tests/test_arraymanipulation.py [hunk 4]
```diff
@@ -605,3 +601,10 @@ def test_2D_array2(self):
...
+@pytest.mark.parametrize("arys", [1.0, dpnp.arange(9.0).reshape(3, 3), [3, 4]])
+def test_atleast_1d(arys):
+    expected = numpy.atleast_1d(arys)
```
Comment (antonwolfy) [inline tests/test_arraymanipulation.py:608]: can we add more tests, probably similar to what numpy does in `numpy/core/tests/test_shape_base.py` for `atleast_1d`?

File: dpnp/dpnp_iface_manipulation.py (hunk 1 above)
Comment (npolina4) [reply dpnp/dpnp_iface_manipulation.py:562]: We can implement support of `casting` and  `dtype` parameters for all functions that use these parameters in separate PR.

File: tests/test_arraymanipulation.py [hunk 5]
```diff
@@ -360,9 +360,9 @@ def test_2D_array(self):
         assert_array_equal(res, desired)
 
     def test_generator(self):
-        with assert_warns(FutureWarning):
+        with pytest.raises(TypeError):
             dpnp.hstack((numpy.arange(3) for _ in range(2)))
```
Comment (antonwolfy) [inline tests/test_arraymanipulation.py:364]:
```suggestion
            dpnp.hstack((dpnp.arange(3) for _ in range(2)))
```

File: tests/test_arraymanipulation.py [hunk 6]
```diff
@@ -360,9 +360,9 @@ def test_2D_array(self):
...
-        with assert_warns(FutureWarning):
+        with pytest.raises(TypeError):
             dpnp.hstack(map(lambda x: x, numpy.ones((3, 2))))
```
Comment (antonwolfy) [inline tests/test_arraymanipulation.py:366]:
```suggestion
            dpnp.hstack(map(lambda x: x, dpnp.ones((3, 2))))
```

**Review by antonwolfy — COMMENTED** [review]
the coverage of `dpnp/dpnp_iface_manipulation.py` is decreased by 1.18 %

File: dpnp/dpnp_iface_manipulation.py [hunk 7]
```diff
@@ -282,7 +282,7 @@ def broadcast_to(array, /, shape, subok=False):
 
 
 def concatenate(
-    arrays, /, *, axis=0, out=None, dtype=None, casting="same_kind", **kwargs
+    arrays, /, *, axis=0, out=None, dtype=None, casting="same_kind"
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_manipulation.py:285]: please remove ``kwargs` also from limitation block

**Review by antonwolfy — APPROVED** [review]
LGTM! Thank you @npolina4

**Pull request merged**